### PR TITLE
Task - Migrate To JSS

### PR DESCRIPTION
**Pull request by hharnisc**
### Purpose

The Buffer Web Components counterpart to: https://github.com/bufferapp/buffer-components/pull/65

### Things to Note

https://github.com/bufferapp/buffer-components/pull/65

I'm pulling in a beta version of Buffer Components with NPM -- it doesn't quite have every change on master. If something looks off let me know!

### Review

This is all pending on https://github.com/bufferapp/buffer-components/pull/65 getting merged to master. So less reviewers on this code.

### CODE REVIEW

**Comment by hharnisc** [comment]
Merging this!

**Comment by djfarrelly** [comment]
hey @hharnisc, instead of making `HoverableFocusable` a thing could we put that as a base component that `Button` extends? It seems a bit verbose to have to use that every time we want a hover action. The code feel a bit harder to follow:

screen shot 2017-05-22 at 12 25 40 pm

**Comment by hharnisc** [comment]
This is a great question @djfarrelly!

There are few things going on here:

**The typical use case for a button just involves wrapping it once**

https://github.com/bufferapp/buffer-web-components/blob/master/DateTimeForm/index.jsx#L48-L55

```jsx
<HoverableFocusable>
  <Button onClick={handleSubmit}>Submit</Button>
</HoverableFocusable>
```

We have clickable text (buttons without style that hold text) that have different `color` and `hoverColor` that are different that the borderless button.


Initially I had created a `HoverableButton` class but then realized we'd have to do this for things like `HoverableLink` and have code duplication. But I think if we just Extended `HoverableFocusable` class we could get best of both worlds here. I'll do a quick followup commit and move that over. Nice one @djfarrelly :+1:

File: ColabTool/MultipleImagesPost/index.jsx [hunk 1]
```diff
@@ -3,9 +3,19 @@ import {
   LinkifiedText,
   MultipleImages,
 } from '@bufferapp/components';
-import style from './style.css';
+// import style from './style.css';
```
Comment (alvarezmelissa87) [inline ColabTool/MultipleImagesPost/index.jsx]: This is so amazing, @hharnisc! 😄  
We can probably remove this comment, I think.

File: ColabTool/PostList/index.jsx [hunk 2]
```diff
@@ -93,6 +97,8 @@ PostList.propTypes = {
   onDeleteClick: PropTypes.func,
   onDeleteConfirmClick: PropTypes.func,
   onEditClick: PropTypes.func,
+  onMoveToDraftsClick: PropTypes.func,
+  onRequestApprovalClick: PropTypes.func,
```
Comment (alvarezmelissa87) [inline ColabTool/PostList/index.jsx]: 👍  Must've missed adding these to propTypes! Thanks for adding them! 😄

File: ColabTool/Shared/PostButtonPanel/style.css [hunk 3]
```diff
@@ -1,11 +0,0 @@
-.button-panel {
```
Comment (alvarezmelissa87) [inline ColabTool/Shared/PostButtonPanel/style.css]: All the 🔴 ! 🙌  😸

File: ColabTool/Shared/PostFooterApproval/story.jsx [hunk 4]
```diff
@@ -3,7 +3,6 @@ import {
   storiesOf,
   action,
 } from '@kadira/storybook';
-import { linkTo } from '@kadira/storybook-addon-links';
```
Comment (alvarezmelissa87) [inline ColabTool/Shared/PostFooterApproval/story.jsx]: 👍

File: HoverableFocusable/index.jsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+export default class HoverableFocusable extends Component {
```
Comment (alvarezmelissa87) [inline HoverableFocusable/index.jsx]: This is so awesome! ❤️  Love having a generic wrapper to make things hoverable/focusable! So cool! 🙌

**Pull request merged**